aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java

```diff
@@ -440,6 +443,42 @@ public void close() throws IOException {
     closeableGroup.close();
   }
 
+  @Override
+  public Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
+    Preconditions.checkArgument(isValidIdentifier(identifier),
+        "Cannot register: invalid table identifier " + identifier);
```

Comment:
`isValidIdentifier` for DynamoDB always returns true. Is this something we can implement (maybe in a separate PR)?